File: multi_process_sp/src/main/java/cn/qingtui/multiprocesssp/lib/RemoteProviderAccessor.java
package cn.qingtui.multiprocesssp.lib;

import java.util.Map;

/**
 * 远程provider 的访问器
 * Date:2019-05-30
 *
 * @author KongMing
 */
public interface RemoteProviderAccessor {

    public <Value> void put(String key, Class<Value> valueClass, String value);

    public <Value> void put(String key, Value value);

    public <Value> Value get(String key, Class<Value> valueClass, Value defaultValue);

    void remove(String key);

    boolean commit();

    void apply();

    Map<String,?> getAll();

    void clear();
}

File: multi_process_sp/src/main/java/cn/qingtui/multiprocesssp/lib/RemoteProviderAccessor.kt
package cn.qingtui.multiprocesssp.lib

/**
 * 远程provider 的访问器
 * Date:2019-05-30
 *
 * @author KongMing
 */
interface RemoteProviderAccessor {
    fun <Value> put(
        key: String,
        valueClass: Class<Value>?,
        value: Value?
    )

    fun <Value> put(key: String, value: Value?)

    operator fun <Value> get(
        key: String,
        valueClass: Class<Value>,
        defaultValue: Value?
    ): Value?

    fun remove(key: String)
    fun commit(): Boolean
    fun apply()
    fun clear()
    fun getAll(): Map<String?, *>?
}